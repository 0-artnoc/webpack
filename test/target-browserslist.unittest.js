const { resolve } = require("../lib/config/browserslistTargetHandler");

describe("browserslist target", () => {
	const tests = [
		// IE
		["ie 11"],

		// Edge
		["edge 79"],

		// Android
		["android 4"],
		["android 4.1"],
		// TODO fix
		// ["android 4.4.3-4.4.4"],

		// Chrome
		["chrome 80"],
		["and_chr 80"],

		// Firefox
		["firefox 68"],
		["and_ff 68"],

		// Opera
		["opera 54"],
		["op_mob 54"],

		// Safari
		["safari 10"],
		["safari TP"],
		["safari 11"],
		["safari 12.0"],
		["safari 12.1"],
		["safari 13"],

		// Samsung
		["samsung 4"],
		["samsung 9.2"],
		// TODO ["samsung 11.1-11.2"]

		// TODO Electron
		// TODO ie_mob
		// TODO Baidu
		// TODO bb
		// TODO op_mini
		// TODO and_uc
		// TODO and_qq
		// TODO kaios
		// TODO unknown for future new browsers

		// Node
		["node 0.10.0"],
		["node 0.12.0"],
		["node 10.0.0"],
		["node 10.17.0"],
		["node 12.19.0"],

		// Multiple
		["firefox 80", "chrome 80"],
		["chrome 80", "node 12.19.0"]
	];

	for (const test of tests) {
		it(`${JSON.stringify(test)}`, () => {
			expect(resolve(test)).toMatchSnapshot();
		});
	}
});
