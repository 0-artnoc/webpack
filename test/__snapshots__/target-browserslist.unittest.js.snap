// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`browserslist target ["chrome 80","node 12.19.0"] 1`] = `
Object {
  "arrowFunction": true,
  "bigIntLiteral": true,
  "browser": null,
  "const": true,
  "destructuring": true,
  "document": null,
  "dynamicImport": false,
  "dynamicImportInWorker": false,
  "electron": false,
  "fetchWasm": null,
  "forOf": true,
  "global": null,
  "globalThis": true,
  "importScripts": false,
  "importScriptsInWorker": true,
  "module": false,
  "node": null,
  "nodeBuiltins": null,
  "nwjs": false,
  "require": null,
  "web": null,
  "webworker": false,
}
`;

exports[`browserslist target ["chrome 80"] 1`] = `
Object {
  "arrowFunction": true,
  "bigIntLiteral": true,
  "browser": true,
  "const": true,
  "destructuring": true,
  "document": true,
  "dynamicImport": true,
  "dynamicImportInWorker": true,
  "electron": false,
  "fetchWasm": true,
  "forOf": true,
  "global": false,
  "globalThis": true,
  "importScripts": false,
  "importScriptsInWorker": true,
  "module": true,
  "node": false,
  "nodeBuiltins": false,
  "nwjs": false,
  "require": false,
  "web": true,
  "webworker": false,
}
`;

exports[`browserslist target ["firefox 80","chrome 80"] 1`] = `
Object {
  "arrowFunction": true,
  "bigIntLiteral": true,
  "browser": true,
  "const": true,
  "destructuring": true,
  "document": true,
  "dynamicImport": true,
  "dynamicImportInWorker": true,
  "electron": false,
  "fetchWasm": true,
  "forOf": true,
  "global": false,
  "globalThis": true,
  "importScripts": false,
  "importScriptsInWorker": true,
  "module": true,
  "node": false,
  "nodeBuiltins": false,
  "nwjs": false,
  "require": false,
  "web": true,
  "webworker": false,
}
`;

exports[`browserslist target ["ie 11"] 1`] = `
Object {
  "arrowFunction": false,
  "bigIntLiteral": false,
  "browser": true,
  "const": true,
  "destructuring": false,
  "document": true,
  "dynamicImport": false,
  "dynamicImportInWorker": false,
  "electron": false,
  "fetchWasm": true,
  "forOf": false,
  "global": false,
  "globalThis": false,
  "importScripts": false,
  "importScriptsInWorker": true,
  "module": false,
  "node": false,
  "nodeBuiltins": false,
  "nwjs": false,
  "require": false,
  "web": true,
  "webworker": false,
}
`;

exports[`browserslist target ["node 0.10.0"] 1`] = `
Object {
  "arrowFunction": false,
  "bigIntLiteral": false,
  "browser": false,
  "const": false,
  "destructuring": false,
  "document": false,
  "dynamicImport": false,
  "dynamicImportInWorker": false,
  "electron": false,
  "fetchWasm": false,
  "forOf": false,
  "global": true,
  "globalThis": false,
  "importScripts": false,
  "importScriptsInWorker": true,
  "module": false,
  "node": true,
  "nodeBuiltins": true,
  "nwjs": false,
  "require": true,
  "web": false,
  "webworker": false,
}
`;

exports[`browserslist target ["node 0.12.0"] 1`] = `
Object {
  "arrowFunction": false,
  "bigIntLiteral": false,
  "browser": false,
  "const": false,
  "destructuring": false,
  "document": false,
  "dynamicImport": false,
  "dynamicImportInWorker": false,
  "electron": false,
  "fetchWasm": false,
  "forOf": true,
  "global": true,
  "globalThis": false,
  "importScripts": false,
  "importScriptsInWorker": true,
  "module": false,
  "node": true,
  "nodeBuiltins": true,
  "nwjs": false,
  "require": true,
  "web": false,
  "webworker": false,
}
`;

exports[`browserslist target ["node 10.0.0"] 1`] = `
Object {
  "arrowFunction": true,
  "bigIntLiteral": false,
  "browser": false,
  "const": true,
  "destructuring": true,
  "document": false,
  "dynamicImport": false,
  "dynamicImportInWorker": false,
  "electron": false,
  "fetchWasm": false,
  "forOf": true,
  "global": true,
  "globalThis": false,
  "importScripts": false,
  "importScriptsInWorker": true,
  "module": false,
  "node": true,
  "nodeBuiltins": true,
  "nwjs": false,
  "require": true,
  "web": false,
  "webworker": false,
}
`;

exports[`browserslist target ["node 10.17.0"] 1`] = `
Object {
  "arrowFunction": true,
  "bigIntLiteral": true,
  "browser": false,
  "const": true,
  "destructuring": true,
  "document": false,
  "dynamicImport": false,
  "dynamicImportInWorker": false,
  "electron": false,
  "fetchWasm": false,
  "forOf": true,
  "global": true,
  "globalThis": false,
  "importScripts": false,
  "importScriptsInWorker": true,
  "module": false,
  "node": true,
  "nodeBuiltins": true,
  "nwjs": false,
  "require": true,
  "web": false,
  "webworker": false,
}
`;

exports[`browserslist target ["node 12.19.0"] 1`] = `
Object {
  "arrowFunction": true,
  "bigIntLiteral": true,
  "browser": false,
  "const": true,
  "destructuring": true,
  "document": false,
  "dynamicImport": false,
  "dynamicImportInWorker": false,
  "electron": false,
  "fetchWasm": false,
  "forOf": true,
  "global": true,
  "globalThis": true,
  "importScripts": false,
  "importScriptsInWorker": true,
  "module": false,
  "node": true,
  "nodeBuiltins": true,
  "nwjs": false,
  "require": true,
  "web": false,
  "webworker": false,
}
`;

exports[`browserslist target ["safari 10"] 1`] = `
Object {
  "arrowFunction": true,
  "bigIntLiteral": false,
  "browser": true,
  "const": true,
  "destructuring": true,
  "document": true,
  "dynamicImport": false,
  "dynamicImportInWorker": false,
  "electron": false,
  "fetchWasm": true,
  "forOf": true,
  "global": false,
  "globalThis": false,
  "importScripts": false,
  "importScriptsInWorker": true,
  "module": false,
  "node": false,
  "nodeBuiltins": false,
  "nwjs": false,
  "require": false,
  "web": true,
  "webworker": false,
}
`;

exports[`browserslist target ["safari 11"] 1`] = `
Object {
  "arrowFunction": true,
  "bigIntLiteral": false,
  "browser": true,
  "const": true,
  "destructuring": true,
  "document": true,
  "dynamicImport": false,
  "dynamicImportInWorker": false,
  "electron": false,
  "fetchWasm": true,
  "forOf": true,
  "global": false,
  "globalThis": false,
  "importScripts": false,
  "importScriptsInWorker": true,
  "module": true,
  "node": false,
  "nodeBuiltins": false,
  "nwjs": false,
  "require": false,
  "web": true,
  "webworker": false,
}
`;

exports[`browserslist target ["safari 12.0"] 1`] = `
Object {
  "arrowFunction": true,
  "bigIntLiteral": false,
  "browser": true,
  "const": true,
  "destructuring": true,
  "document": true,
  "dynamicImport": true,
  "dynamicImportInWorker": true,
  "electron": false,
  "fetchWasm": true,
  "forOf": true,
  "global": false,
  "globalThis": false,
  "importScripts": false,
  "importScriptsInWorker": true,
  "module": true,
  "node": false,
  "nodeBuiltins": false,
  "nwjs": false,
  "require": false,
  "web": true,
  "webworker": false,
}
`;

exports[`browserslist target ["safari 12.1"] 1`] = `
Object {
  "arrowFunction": true,
  "bigIntLiteral": false,
  "browser": true,
  "const": true,
  "destructuring": true,
  "document": true,
  "dynamicImport": true,
  "dynamicImportInWorker": true,
  "electron": false,
  "fetchWasm": true,
  "forOf": true,
  "global": false,
  "globalThis": true,
  "importScripts": false,
  "importScriptsInWorker": true,
  "module": true,
  "node": false,
  "nodeBuiltins": false,
  "nwjs": false,
  "require": false,
  "web": true,
  "webworker": false,
}
`;

exports[`browserslist target ["safari 13"] 1`] = `
Object {
  "arrowFunction": true,
  "bigIntLiteral": false,
  "browser": true,
  "const": true,
  "destructuring": true,
  "document": true,
  "dynamicImport": true,
  "dynamicImportInWorker": true,
  "electron": false,
  "fetchWasm": true,
  "forOf": true,
  "global": false,
  "globalThis": true,
  "importScripts": false,
  "importScriptsInWorker": true,
  "module": true,
  "node": false,
  "nodeBuiltins": false,
  "nwjs": false,
  "require": false,
  "web": true,
  "webworker": false,
}
`;

exports[`browserslist target ["safari TP"] 1`] = `
Object {
  "arrowFunction": true,
  "bigIntLiteral": true,
  "browser": true,
  "const": true,
  "destructuring": true,
  "document": true,
  "dynamicImport": true,
  "dynamicImportInWorker": true,
  "electron": false,
  "fetchWasm": true,
  "forOf": true,
  "global": false,
  "globalThis": true,
  "importScripts": false,
  "importScriptsInWorker": true,
  "module": true,
  "node": false,
  "nodeBuiltins": false,
  "nwjs": false,
  "require": false,
  "web": true,
  "webworker": false,
}
`;
